docroot/sites/iwp.uiowa.edu/modules/iwp_core/iwp_core.module

```diff
@@ -159,9 +217,76 @@ function iwp_core_resource_years_allowed_values(FieldStorageConfig $definition,
 
   $i = $thisyear + 10;
   $options = [];
-  while ($i >= $thisyear - 50) :
+  while ($i >= 1950) :
     $options[$i] = $i;
     $i--;
   endwhile;
   return $options;
 }
+
+/**
+ *
+ */
+function iwp_core_entity_extra_field_info() {
+  $extra = [];
+
+  // Resource button.
+  $extra['node']['resource']['display']['field_resource_button'] = [
+    'label' => t('Resource button'),
+    'description' => t('The button to link to the designate resource.'),
+    'weight' => 1,
+    'visible' => TRUE,
+  ];
+
+  return $extra;
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function iwp_core_node_view(array &$build, EntityInterface $node, EntityViewDisplayInterface $display, $view_mode) {
+  if ($node->bundle() === 'resource' && $view_mode === 'full') {
+
+    // Resource button.
+    if ($display->getComponent('field_resource_button')) {
+      $link = '';
+
+      $node_type = $node->bundle();
+      if ($node_type == 'resource' && isset($node->get('field_resource_type')->getValue()[0]['value'])) {
+        $resource_type = $node->get('field_resource_type')->getValue()[0]['value'];
+
+        if ($resource_type === 'video' || $resource_type === 'document') {
+          $media_id = $node->get('field_resource_media')->getValue()[0]['target_id'];
+          $media = Media::load($media_id);
+
+          if ($media->bundle() === 'file') {
+            $fid = $media->getSource()->getSourceFieldValue($media);
+            $file = File::load($fid);
+
+            $link = $file->getFileUri();
+          }
+          elseif ($media->bundle() === 'remote_video') {
+            $link = $media->getSource()->getSourceFieldValue($media);
+          }
+        }
+      }
+
+      if (!empty($link)) {
+        $build['field_resource_button'] = [
+          '#type' => 'markup',
```

Comment:
I haven't pulled down, but is this doing something that we can't achieve with `'#type' => 'link',` instead of markup?

Reply:
starting to see how this could get complicated, but just trying to avoid one-off hardcoded things that would need to be tracked down later for some reason.

Reply:
I tried `'#type' => 'link'` and I remember having difficulty getting everything to work. I can try again.

Reply:
copy/paste from a Slack DM:

thinking more of invoking the field template as a template suggestion and preprocessing the field to have the same render array as an actual link field.
```
$variables['items'][0] = [
      'content' => [
        '#type' => 'link',
...
```
if you want to move forward with what you have, i understand. i don't have the time right now to pull down and poke

Reply:
also as i mentioned in Slack, I can't point to an existing example of this so I am not quite sure it is even possible to be an extra field but invoke all of field type's formatter goodies.

used to be [pseudo_field](https://www.drupal.org/project/pseudo_field) which could mimic the field markup.

Reply:
FWIW, I went down a rabbit hole trying to mimic field output without it being an actual field. The link part is coming up short...

```php
       $link = [
          '#type' => 'link',
          '#title' => 'View source',
          '#url' => Url::fromUri($link),
        ];
        $build['field_resource_button'] = [
          '#theme' => 'field',
          '#title' => t('Resource source'),
          '#label_display' => 'visually_hidden',
          '#field_type' => 'link',
          '#formatter' => 'link',
          '#field_name' => 'field_resource_button',
          '#entity_type' => 'node',
          '#bundle' => 'resource',
          '#view_mode' => 'default',
          '#is_multiple' => FALSE,
          '#items' => [
            'content' => [$link],
          ],
        ];
```

<img width="583" alt="Screenshot 2024-06-27 at 3 58 50 PM" src="https://github.com/uiowa/uiowa/assets/4663676/2629fda4-1109-4b79-b680-831db7ffbe22">

Reply:
I got your suggestion to work in https://github.com/uiowa/uiowa/pull/7845/commits/ea3f67ed74fcfaab5ed1b6287d078e829b7ac266.  Thanks for going down the rabbit hole!

Reply:
mind = absolutely blown. 🤯  Awesome @bspeare!